File: java/palindromecheck.java
import java.util.Scanner;
public class palindromecheck
{
    public static void main(String[] args)
    {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int rev=0;
        int temp=n;
        while(n > 0)    
        {
            int rem = n % 10;
            rev = rev * 10 + rem;
            n = n / 10;
        }
        if(temp==rev)
        {
            System.out.println("palindrome");
        }
        else
        {
            System.out.println("not a palindrome");
        }
    }
}

File: cpp/palindromecheck.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    int n;
    cin>>n;
    int temp;
    temp=n;
    int rev=0;
    while(n>0)
    {
        rev=rev*10+n%10;
        n=n/10;
    }
    if(temp==rev)
    {
        cout<<"palindrome";
    }
    else
    {
        cout<<"not palindrome";
    }
}

//Tc=O(Logn) SC=O(1)